test/jdk/tools/jpackage/helpers/jdk/jpackage/test/JPackageCommand.java

```diff
@@ -298,6 +299,57 @@ public JPackageCommand setFakeRuntime() {
         return this;
     }
 
+    public void createJPackageXMLFile(String mainLauncher, String mainClass)
+            throws IOException {
+        String appImage = getArgumentValue("--app-image");
+        if (appImage == null) {
+            throw new RuntimeException("Error: --app-image expected");
+        }
```

Comment:
Can be simplified to:
```
Path jpackageXMLFile = AppImageFile.getPathInAppImage(
                Optional.ofNullable(getArgumentValue("--app-image")).map(
                        Path::of).orElseThrow(() -> {
                            return new RuntimeException(
                                    "Error: --app-image expected");
                        }));
```

And you don't need explicit `ApplicationLayout.platformAppImage()` call below.

Reply:
Done. I also switched to use AppImageFile.getVersion() and AppImageFile.getPlatform().